fvm/fvm.go

```diff
@@ -141,7 +125,21 @@ func (vm *VirtualMachine) Run(
 			WithMaxInteractionSizeAllowed(interactionLimit),
 	)
 
-	return proc.Run(ctx, txnState, txnPrograms)
+	err = proc.Run(ctx, txnState, txnPrograms)
+	if err != nil {
+		return err
+	}
+
+	// Note: it is safe to skip committing the parsed programs for non-normal
+	// transactions (i.e., bootstrap and script) since these do not invalidate
+	// programs.
+	if proc.Type() == TransactionProcedureType {
+		// NOTE: It is not safe to ignore txnPrograms' commit error for
+		// transactions that trigger programs invalidation.
+		return txnPrograms.Commit()
```

Comment:
for understanding purpose, how should we handle the error from `Commit()` here, is the current error handling enough for this new error source?

Reply:
this will halt the execution node.